src/redprl/refiner.fun

```diff
@@ -954,11 +954,9 @@ struct
           case catjdg of
              CJ.TRUE ty => ty
            | _ => raise E.error [E.% "Equality.Hyp: expected truth hypothesis"]
+        val (goalTy, _) = makeGoal @@ (I, H) >> CJ.EQ_TYPE (ty, ty')
```

Comment:
The reason we should force `hyp` to use alpha equivalence rather than making a new subgoal is primarily that we want `hyp` to fail synchronously (in the proof theoretic sense). This is better for automation.

The behavior you want can be emulated in a tactic.